File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    private static final int[] dx = {1, 0, -1};
    private static final int[] dy = {0, 1, -1};
    
    public int[] solution(int n) {
        int[][] triangle = new int[n][n];
        
        int x = -1;
        int y = 0;
        int count = 1;
        
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j++) {
                x += dx[i % 3];
                y += dy[i % 3];
                triangle[x][y] = count++;
            }
        }
        
        List<Integer> answer = new ArrayList<>();
        for (int[] row : triangle) {
            for (int r : row) {
                if (r != 0) {
                    answer.add(r);
                }
            }
        }
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    answer = [[0] * i for i in range(1, n + 1)]
    
    dx, dy = (1, 0, -1), (0, 1, -1)
    x, y = -1, 0
    count = 1
    
    for i in range(n):
        for j in range(i, n):
            x, y = x + dx[i % 3], y + dy[i % 3]
            answer[x][y] = count
            count += 1
    
    return [k for row in answer for k in row]
